filesystem/fat32/fat32.go

```diff
@@ -562,14 +571,87 @@ func (fs *FileSystem) OpenFile(p string, flag int) (filesystem.File, error) {
 	}, nil
 }
 
-// Label get the label of the filesystem
+// Label get the label of the filesystem from the secial file in the root directory.
+// The label stored in the boot sector is ignored to mimic Windows behavior which
+// only stores and reads the label from the special file in the root directory.
 func (fs *FileSystem) Label() string {
-	// be sane about everything existing
+	// locate the filesystem root directory
```

Comment:
All of this reads _solely_ from the volume label magic file on the filesystem, and drops reading the volume label on the boot sector. 

Is there a compatibility issue? Are there implementations that write solely there, and we will miss it? Should we read from magical file and fall back to volume label on boot sector, or is that wrong? Conversely, is there a separate construct for that label that we could use?